packages/backend-for-frontend/src/services/catalogService.ts

```diff
@@ -1786,5 +1786,27 @@ export function catalogServiceBuilder(
 
       return { id: descriptorId };
     },
+    async checkEServiceNameAvailability(
+      name: string,
+      { headers, logger, authData }: WithLogger<BffAppContext>
+    ): Promise<boolean> {
+      logger.info(
+        `Checking e-service name availability ${name} for producer ${authData.organizationId}`
+      );
+
+      const templateInstances = await getAllFromPaginated((offset, limit) =>
+        catalogProcessClient.getEServices({
+          headers,
+          queries: {
+            limit,
+            offset,
+            producersIds: [authData.organizationId],
+            name,
+          },
+        })
+      );
+
+      return !templateInstances.some((e) => e.name === name);
```

Comment:
```suggestion
      return templateInstances.length === 0;
```
Shouldn't this be enought? Since we are retrieving specifically the eservices with that name, if it returns something then the name is already taken

Reply:
No, it is not, q is not an exact match.
For example, I could have two e-services:

1. "pippo23"
2. "pippo24"

I want to check if I can create an e-service named "pippo2".
If I apply your logic the service would return `false` even though I can.

Reply:
> No, it is not, the `q` filter does not apply an exact match. For example, I could have two e-services:
> 
>     1. "pippo23"
> 
>     2. "pippo24"
> 
> 
> I want to check if I can create an e-service named "pippo2". If I apply your logic the service would return `false` even though I can create the e-service with that name.

You're right but then we should do a case insensitive comparison like just to be sure

```typescript
return !eservices.some((e) => e.name.toLowerCase() === name.toLowerCase());
```